### DISCUSSION TITLE
Communication with ROS2 running in another device in the same network

### DISCUSSION THREAD

**Discussion by kinrep71** [noelex/rclnet#6] [Q&A]
Hi all,

is it possible to use rclnet for communication with ROS2 running in another device which is in the same network as the system where my .NET application runs? I have a ROS2 Humble running in docker container which runs in  Linux OS in IOT device. I can access the device using SSH.

Thank You in advance

**Comment by noelex** [comment]
The question isn't really about ROS 2 or rclnet. You should check out documentation about [Docker networking](https://docs.docker.com/network/).

To answer your question, yes it's possible and there are several ways to achieve this. Pick one which you consider convenient:

1. ~~Run .NET application inside a container within the same Docker network of ROS 2 container.~~
2. Run ROS 2 container in host network mode and run .NET application in the host OS.

Considering that you want to communicate with ROS 2 running on another machine, this leaves you only the second option.
